server/matchmaker/search.py

```diff
@@ -110,26 +111,25 @@ def search_expansion(self) -> float:
         Defines how much to expand the search range of game quality due to waiting
         time.
 
-        The graph of this function over time looks essentially like this:
-                           END (x)-> ___ <- MAX (y)
-                                    /
-                                ___/ <- START (x)
-        The search threshold will not expand until a certain time START has been
-        reached. Then it will expand linearly with time until time END, at which
-        point it will have reached it's maximum value and will not expand
-        further.
-        """
-        elapsed = time.time() - self.start_time
-        MAX = config.LADDER_SEARCH_EXPANSION_MAX
-        START = config.LADDER_SEARCH_EXPANSION_START
-        END = config.LADDER_SEARCH_EXPANSION_END
-
-        if elapsed < START:
-            return 0.0
-        if elapsed > END:
-            return MAX
-
-        return (MAX / (END - START)) * (elapsed - START)
+        The threshold will expand linearly with every failed matching attempt
+        until it reaches the specified MAX.
+        """
+        
+        return min(
+                self._failed_matching_attempts * config.LADDER_SEARCH_EXPANSION_STEP,
+                config.LADDER_SEARCH_EXPANSION_MAX
+        )
+
+    def register_failed_matching_attempt(self) -> int:
+        """
+        Signal that matchmaker tried to match this search but was unsuccessful
+        and increase internal counter by one.
+        Returns number of failed matching attempts after increase.
+        """
+
+        self._failed_matching_attempts += 1
+        return self._failed_matching_attempts
```

Comment:
Not a fan of the return value, especially because it's just used for logging. I'd say just access the value directly when you need it. That way there is no confusion as to whether the return value is the value before increment or after. It will be completely clear.